test/kubernetes_unittest.cc

```diff
@@ -137,4 +145,58 @@ TEST_F(KubernetesTest, ComputePodAssociations) {
       ComputePodAssociations(reader, pod->As<json::Object>());
   EXPECT_EQ(expected_associations->ToString(), associations->ToString());
 }
+
+TEST_F(KubernetesTest, GetPodMetadata) {
+  Configuration config(std::stringstream(
+    "KubernetesClusterName: TestClusterName\n"
+    "KubernetesClusterLocation: TestClusterLocation\n"
+    "MetadataApiResourceTypePerarator: \",\"\n"
+    "MetadataIngestionRawContentVersion: TestVersion\n"
+  ));
+  Environment environment(config);
+  KubernetesReader reader(config, nullptr);  // Don't need HealthChecker.
+
+  const auto m = GetPodMetadata(
+      reader, json::object({
+        {"metadata", json::object({
+          {"namespace", json::string("TestNamespace")},
+          {"name", json::string("TestName")},
+          {"uid", json::string("TestUid")},
+          {"creationTimestamp", json::string("2018-03-03T01:23:45.678901234Z")},
+        })},
+      })->As<json::Object>(),
```

Comment:
These `->As<json::Object>()` and `->ToString()` calls get lost at the end of inline JSON values... That's why I'm suggesting factoring them out into locals, to make this more readable... WDYT?

Reply:
I agree, I'll go forward and do this now.